code/datums/components/drift.dm

```diff
@@ -192,3 +216,52 @@
 		return
 	if(world.time < block_inputs_until)
 		return COMSIG_MOB_CLIENT_BLOCK_PRE_MOVE
+
+/datum/component/drift/proc/attempt_halt(mob/source, movement_dir, continuous_move, atom/backup)
+	SIGNAL_HANDLER
+
+	if (get_dir(source, backup) == movement_dir || source.loc == backup.loc)
+		if (drift_force >= INERTIA_FORCE_THROW_FLOOR)
+			source.throw_at(backup, 1, round(1 + (drift_force - INERTIA_FORCE_THROW_FLOOR) / INERTIA_FORCE_PER_THROW_FORCE), spin = FALSE)
```

Comment:
similar

Reply:
also you know what round(thing) does right? do you MEAN FLOOR() or floor() can never recall?

Reply:
This converts force to throwforce, also nonlinearly starting from 7 and upping force by one every 4.5 drift_force. This way moths dont bash into walls but functional wing users and jetpacks do.
![image](https://github.com/user-attachments/assets/ab16ebfb-695d-4faa-994e-004261620963)

Reply:
ditto

Reply:
also I take it you did intend floor() here?